distributed_shampoo/distributed_shampoo.py

```diff
@@ -474,6 +479,7 @@ def __init__(
         self._shampoo_pt2_compile_config: Optional[ShampooPT2CompileConfig] = (
             shampoo_pt2_compile_config
         )
+        self._experimental_param_to_lr = experimental_param_to_lr
```

Comment:
In order to handle support for this properly, we will need to create a function that constructs this mapping automatically from each parameter (within each parameter group) to its learning rate and modify the parameter groups defined typically by `torch.optim.Optimizer` (see https://pytorch.org/docs/stable/_modules/torch/optim/optimizer.html#Optimizer). 

This may need to be moved prior to the `super().__init__()` function above, so we can call this on the parameter groups.